File: java/src/test/java/com/ter/adapters/DotEnvTest.java
package com.ter.adapters;

import org.junit.Test;
import static org.junit.Assert.*;
import java.util.Map;

public class DotEnvTest {
    @Test
    public void testParseSimple() {
        String content = "KEY1=value1\nKEY2=value2";
        Map<String, Object> result = DotEnv.parse(content);

        assertEquals("value1", result.get("KEY1"));
        assertEquals("value2", result.get("KEY2"));
    }

    @Test
    public void testParseIgnoresComments() {
        String content = "# Comment\nKEY=value\n# Another";
        Map<String, Object> result = DotEnv.parse(content);

        assertEquals(1, result.size());
        assertEquals("value", result.get("KEY"));
    }

    @Test
    public void testParseQuoted() {
        String content = "KEY1=\"quoted value\"\nKEY2='single'";
        Map<String, Object> result = DotEnv.parse(content);

        assertEquals("quoted value", result.get("KEY1"));
        assertEquals("single", result.get("KEY2"));
    }

    @Test
    public void testParseHeredoc() {
        String content = "CERT=<<EOF\nBEGIN CERT\nEOF";
        Map<String, Object> result = DotEnv.parse(content);

        assertTrue(((String) result.get("CERT")).contains("BEGIN CERT"));
    }

    @Test
    public void testGenerate() {
        Map<String, Object> values = Map.of("KEY1", "value1", "KEY2", "value2");
        String result = DotEnv.generate(values);

        assertTrue(result.contains("KEY1=value1"));
        assertTrue(result.contains("KEY2=value2"));
    }

    @Test
    public void testEscapeValue() {
        String escaped = DotEnv.escapeValue("value with \"quotes\"");
        assertTrue(escaped.contains("\\\""));

        String escaped2 = DotEnv.escapeValue("simple");
        assertEquals("simple", escaped2);
    }

    @Test
    public void testUnescapeValue() {
        String unescaped = DotEnv.unescapeValue("\"quoted value\"");
        assertEquals("quoted value", unescaped);

        String unescaped2 = DotEnv.unescapeValue("unquoted");
        assertEquals("unquoted", unescaped2);
    }

    @Test
    public void testRoundTrip() {
        Map<String, Object> original = Map.of(
            "KEY1", "value1",
            "KEY2", "value2"
        );

        String generated = DotEnv.generate(original);
        Map<String, Object> parsed = DotEnv.parse(generated);

        assertEquals("value1", parsed.get("KEY1"));
        assertEquals("value2", parsed.get("KEY2"));
    }
}

File: rust/tests/dotenv_test.rs
use ter::adapters::DotEnv;
use std::collections::HashMap;

#[test]
fn test_parse_simple() {
    let content = "KEY1=value1\nKEY2=value2";
    let result = DotEnv::parse(content);

    assert_eq!(result.get("KEY1").unwrap(), "value1");
    assert_eq!(result.get("KEY2").unwrap(), "value2");
}

#[test]
fn test_parse_comments() {
    let content = "# Comment\nKEY=value\n# Another";
    let result = DotEnv::parse(content);

    assert_eq!(result.len(), 1);
    assert_eq!(result.get("KEY").unwrap(), "value");
}

#[test]
fn test_parse_quoted() {
    let content = "KEY1=\"quoted value\"\nKEY2='single'";
    let result = DotEnv::parse(content);

    assert_eq!(result.get("KEY1").unwrap(), "quoted value");
    assert_eq!(result.get("KEY2").unwrap(), "single");
}

#[test]
fn test_escape_value() {
    let escaped = DotEnv::escape_value("value with \"quotes\"");
    assert!(escaped.contains("\\\""));

    let simple = DotEnv::escape_value("simple");
    assert_eq!(simple, "simple");
}

#[test]
fn test_unescape_value() {
    let unescaped = DotEnv::unescape_value("\"quoted value\"");
    assert_eq!(unescaped, "quoted value");

    let unquoted = DotEnv::unescape_value("unquoted");
    assert_eq!(unquoted, "unquoted");
}

#[test]
fn test_generate() {
    let mut values = HashMap::new();
    values.insert("KEY1".to_string(), "value1".to_string());
    values.insert("KEY2".to_string(), "value2".to_string());

    let result = DotEnv::generate(&values);
    assert!(result.contains("KEY1=value1"));
    assert!(result.contains("KEY2=value2"));
}
